src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public SQLXML createSQLXML() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Database meta data
+
+    @Override
+    public DatabaseMetaData getMetaData() throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-37");
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getCatalog() {
+        return null;
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getSchema() {
+        return null;
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // unsupported operations
+
+    // stored procedure equivalence in MongoDB won't be supported
+    // see https://www.mongodb.com/resources/products/capabilities/stored-procedures
+
+    @Override
+    public CallableStatement prepareCall(String mql) throws SQLException {
+        throw new UnsupportedOperationSQLException("Stored procedure equivalence in MongoDB unsupported");
+    }
+
+    @Override
+    public CallableStatement prepareCall(String mql, int resultSetType, int resultSetConcurrency) throws SQLException {
+        throw new UnsupportedOperationSQLException("Stored procedure equivalence in MongoDB unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // dummy implementations
+
+    /**
+     * Only used in {@link org.hibernate.engine.jdbc.spi.SqlExceptionHelper}.
+     *
+     * <p>Currently no need arises to record warning in this connection class.
+     */
+    @Override
+    public @Nullable SQLWarning getWarnings() {
+        return null;
+    }
+
+    /** Only used in {@link org.hibernate.engine.jdbc.spi.SqlExceptionHelper}. */
+    @Override
+    public void clearWarnings() {
+        // no-op
+    }
+
+    /** Only used in {@link org.hibernate.dialect.OracleArrayJdbcType#getBinder(JavaType)} */
+    @Override
+    public <T> T unwrap(Class<T> unwrapType) {
+        throw new UnknownUnwrapTypeException(unwrapType);
+    }
+}
```

Comment:
The above method comes from the common `Wrapper` parent interfaces, but it is only ever used in OracleArrayJdbcType#getBinder() to invoke driver-specific customization method in lieu of JDBC method. Well, I don't think we need to implement it for only Oracle Dialect might use it.